src/nwb/NWBRecording.cpp

```diff
@@ -16,14 +16,14 @@ NWBRecording::~NWBRecording()
   }
 }
 
-Status NWBRecording::openFile(const std::string& rootFolder,
-                              const std::string& baseName,
-                              int experimentNumber,
+Status NWBRecording::openFile(const std::string& filename,
                               std::vector<Types::ChannelVector> recordingArrays,
                               const std::string& IOType)
 {
-  std::string filename =
-      rootFolder + baseName + std::to_string(experimentNumber) + ".nwb";
+  // close any existing files
+  if (nwbfile != nullptr) {
+    nwbfile->finalize();
```

Comment:
Why not call ``this->closeFile()`` here instead?

Reply:
Good point, not sure why I didn't do that, updated.